### PR TITLE
Adds typescript definitions

### PR DESCRIPTION
**Pull request by hoefling**
Great library, the only thing I'm personally missing is the Typescript support. This PR adds a type definition file so `using` can be imported in Typescript spec files.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Type definitions for jasmine-data-provider 2.2.0
+// Project: https://github.com/MortalFlesh/jasmine-data-provider
+// Definitions by: Oleg Hoefling <https://github.com/hoefling>
+
+declare interface DataDescProvider<T> {
+  [key: string]: T;
+}
+
+declare function using<T>(data: T[], specDefinitions: (datum: T) => void): void;
```
Comment (MortalFlesh) [inline index.d.ts:9]: I'm bit confused from the `datum: T` - what does it mean? I think `item: T` or `case: T` would be more understandable..

Comment (MortalFlesh) [inline index.d.ts:9]: I'd rather don't use acronyms - especially in the declarations and definitions - `specDefinitions`, `DataDescProvider`, ...

Concrete and full names would be more explicit and more readable .. so for example something like
```typescript
ProviderWithDescription<T> {
    [description: string]: T;
}
```
or
```typescript
NamedDataProvider<T> {
    [caseName: string]: T;
}
```
or whatever makes more sense..

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+declare function using<T>(dataProvider: () => T[], specDefinitions: (datum: T) => void): void;
```
Comment (MortalFlesh) [inline index.d.ts:10]: could the duplicated definition (type?) be abstracted somehow?
`specDefinitions: (datum: T) => void`

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// Type definitions for jasmine-data-provider 2.2.0
```
Comment (MortalFlesh) [inline index.d.ts:1]: Is the version here necessary? And should I increase this version with every release?

**Review by MortalFlesh — CHANGES_REQUESTED** [review]
Hello,

first of all thanks for this PR :+1: 

I personally don't use TS, I'm more Flow type user - so I don't have much experience with those declarations - so if my comments don't make sense, please just correct me :)

What about some example of usage this typescript definiton (of each one) - could you add it to readme? I guess it would help.

File: index.d.ts (hunk 3 above)
Comment (MartinKristof) [reply index.d.ts:1]: Maybe reference is missing. https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html

**Comment by tlaziuk** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29772

sorry, I didn't notice that you are already working on that

**Comment by MortalFlesh** [comment]
@tlaziuk It's ok I guess.

As I mentioned before - I'm not really a TS guy, so .. does this mean, that if the definition is already merged in `DefinitelyTyped`, this PR is redundant?

Or could the definition be both here and there - should it be the same in that case?

**Comment by hoefling** [comment]
@tlaziuk cool, I guess this is obsolete then! I started adding the typedefs here only because I couldn't find the `@types/jasmine-data-provider` first ;-)

@MortalFlesh I'd say this PR is superseded and can be closed! The solution with typedefs in `DefinitelyTyped` is a better one IMO - if one needs the Typescript definitions, `npm install @types/jasmine-data-provider` adds them to a project and you are not forced to pollute the package with the code you don't want to maintain.

**Pull request closed without merging**